File: generator/example_pool/python/schemas/s3/s3_output.py
class Vehicle: 
    factories = {}

    def __init__(self, id, price):
        self.id = id;
        self.price = price;

    @classmethod
    def fromJson(cls, json):
        if json['vehicleType'] != None:
            return cls.factories[json['vehicleType']](json)
        return Vehicle(json['id'], json['price'])

    def toJson(self):
        return self.__dict__

    @staticmethod
    def register_factory(vehicleType, factoryFunction):
        Vehicle.factories[vehicleType] = factoryFunction

    def __str__(self):
        return f'Vehicle(id={self.id}, price={self.price})'

    def __eq__(self, other):
        if not isinstance(other, Vehicle):
            return False

        return (self.id == other.id) and (self.price == other.price)

File: generator/example_pool/js/schemas/s3/s3_output.js
class Vehicle {
    constructor(id, price) {
        this.id = id;
        this.price = price;
    }

    static fromJson(json) {
        if (json.documentType) {
            return Vehicle.factories[json.documentType](json);
        }
        return new Vehicle(json.id, json.price);
    }

    toJson() {
        return JSON.stringify(this);
    }
}

Vehicle.factories = {};

Vehicle.registerFactory = function (type, factoryFunction) {
    Vehicle.factories[type] = factoryFunction;
};

export default Vehicle;
